pydicom/tag.py

```diff
@@ -166,16 +167,24 @@ def __ge__(self, other: object) -> bool:
 
     def __gt__(self, other: object) -> bool:
         """Return ``True`` if `self` is greater than `other`."""
+        # Check if comparing with another Tag object; if not, create a temp one
+        if not isinstance(other, BaseTag):
+            try:
+                other = Tag(other)
+            except Exception:
+                raise TypeError("Cannot compare Tag with non-Tag item")
+
         return not (self == other or self < other)
 
     def __eq__(self, other: object) -> bool:
         """Return ``True`` if `self` equals `other`."""
         # Check if comparing with another Tag object; if not, create a temp one
-        if not isinstance(other, int):
+        if not isinstance(other, BaseTag):
```

Comment:
`int` is broader type so it will be more efficient since we won't be doing unecessary `Tag(int)` calls (which are relatively expensive).

The type in `__lt__()` should be `int` rather than `BaseTag`.

Reply:
In both `__lt__` and `__eq__`, I changed to `int` type